x-pack/plugin/rank-rrf/src/main/java/org/elasticsearch/xpack/rank/rrf/RRFRankBuilder.java

```diff
@@ -36,16 +36,17 @@
 
 /**
  * The builder to support RRF. Adds user-defined parameters for window size and rank constant.
+ *
+ * @deprecated RRF support is provided through the retriever framework. Please use {@link RRFRetrieverBuilder instead}
  */
+@Deprecated
 public class RRFRankBuilder extends RankBuilder {
```

Comment:
All rank builders are deprecated. WE should have none of them right? Deprecate `RankBuilder` right?

Reply:
Well, we currently still have the `TextSimilarityRankBuilder` for the semantic reranker, which however was never exposed as a user-facing API through `rank`, but just through the `text_similarity_rerank` retriever. 

Once, we're clear to remove some of the (now) dead code, the distinction of concepts will be much more clear.

Reply:
++